apps/condo/domains/common/components/Pickers/DatePicker.tsx

```diff
@@ -1,7 +1,45 @@
+import React, { useContext } from 'react'
 import { Dayjs } from 'dayjs'
 import dayjsGenerateConfig from 'rc-picker/lib/generate/dayjs'
-import generatePicker from 'antd/lib/date-picker/generatePicker'
+import generatePicker, {
+    PickerDateProps,
+    PickerProps,
+    PickerTimeProps,
+    RangePickerProps,
+} from 'antd/lib/date-picker/generatePicker'
+import { ConfigProvider } from 'antd'
+
+type PickerType =
+    React.ComponentClass<PickerProps<Dayjs>, any>
+    | React.ComponentClass<RangePickerProps<Dayjs>>
+    | React.ComponentClass<Omit<PickerDateProps<Dayjs>, 'picker'>, any>
+    | React.ComponentClass<Omit<PickerTimeProps<Dayjs>, 'picker'>, any>
+
+type PropsWithRef<P, R = any> = P & { ref?: React.ForwardedRef<R> }
+
+type MergedDatePickerType = React.FC<PropsWithRef<PickerProps<Dayjs>>> & {
+    WeekPicker: React.FC<PropsWithRef<Omit<PickerDateProps<Dayjs>, 'picker'>>>,
+    MonthPicker: React.FC<PropsWithRef<Omit<PickerDateProps<Dayjs>, 'picker'>>>,
+    YearPicker: React.FC<PropsWithRef<Omit<PickerDateProps<Dayjs>, 'picker'>>>,
+    RangePicker: React.FC<PropsWithRef<RangePickerProps<Dayjs>>>,
+    TimePicker: React.FC<PropsWithRef<Omit<PickerTimeProps<Dayjs>, 'picker'>>>,
+    QuarterPicker: React.FC<PropsWithRef<Omit<PickerTimeProps<Dayjs>, 'picker'>>>,
+}
 
 const DatePicker = generatePicker<Dayjs>(dayjsGenerateConfig)
 
-export default DatePicker
\ No newline at end of file
+const generateDatepickerWithLocale = (Picker: PickerType) => (props) => {
+    const { locale } = useContext(ConfigProvider.ConfigContext)
+
+    return <Picker locale={locale.DatePicker} {...props} />
+}
+
+const MergedDatePicker = generateDatepickerWithLocale(DatePicker) as MergedDatePickerType
+MergedDatePicker.WeekPicker = generateDatepickerWithLocale(DatePicker.WeekPicker)
```

Comment:
You can’t call hooks outside of react component: https://reactjs.org/docs/hooks-rules.html